### PR TITLE
Release v0.5.0 doc validation

### PR DESCRIPTION
**Pull request by kevin85421**
## Why are these changes needed?

1. This PR mainly focuses on correctness validation rather than document improvement.
2. Use the nightly KubeRay operator image to test the documents.

### Sample YAML files
```sh
RAY_IMAGE=rayproject/ray:2.3.0 OPERATOR_IMAGE=kuberay/operator:nightly python3 tests/test_sample_raycluster_yamls.py 2>&1 | tee log
```
<details>
<summary>Screenshot</summary>
Screen Shot 2023-03-29 at 2 58 46 PM
</details>

   - [x] ray-cluster.ingress.yaml
   - [x] ray-cluster.mini.yaml
   - [x] ray-cluster.external-redis.yaml
   - [x] ray-cluster.heterogeneous.yaml
   - [x] ray-cluster.complete.yaml
   - [x] ray-cluster.tls.yaml
   - [x] ray-cluster.autoscaler.yaml
   - [x] ray-cluster.head-command.yaml
   - [x] ray-cluster.separate-ingress.yaml

```sh
RAY_IMAGE=rayproject/ray:2.3.0 OPERATOR_IMAGE=kuberay/operator:nightly python3 tests/test_sample_rayservice_yamls.py 2>&1 | tee log
```

<details>
<summary>Screenshot</summary>
Screen Shot 2023-03-29 at 5 03 04 PM
</details>

- [x] ray_v1alpha1_rayservice.yaml
    * Since we have chosen to make the entire Kubernetes Service configurable (https://github.com/ray-project/kuberay/issues/877) to users, this PR removes `headServiceAnnotations` from the YAML.

```sh
# Step 0: Create a GKE Kubernetes cluster with autopilot on GCP

# Step 1: Install KubeRay operator (path: helm-chart/kuberay-operator)
# I got the error "Error: INSTALLATION FAILED: create: failed to create: Request entity too large: limit is 3145728" when I use `helm install kuberay-operator .`
helm template kuberay-operator . | kubectl apply -f -

# Step 2: Create a RayCluster with large Ray Pods (14 CPUs & 54 GB RAM / Pod).
kubectl apply -f ray-cluster.complete.large.yaml

# Step 3: Delete the RayCluster
kubectl delete -f ray-cluster.complete.large.yaml

# Step 4: Create an autoscaling RayCluster with large Ray Pods (14 CPUs & 54 GB RAM / Pod).
kubectl apply -f ray-cluster.autoscaler.large.yaml

# Step 5: Scale up the RayCluster
# We currently have 28 CPUs in the RayCluster (1 head + 1 worker). Hence, we request 29 CPUs, and expect the autoscaler will scale up the cluster.
export HEAD_POD=$(kubectl get pods -o custom-columns=POD:metadata.name | grep raycluster-autoscaler-head)
kubectl exec $HEAD_POD -it -c ray-head -- python -c "import ray;ray.init();ray.autoscaler.sdk.request_resources(num_cpus=29)"
```

<details>
<summary>ray-cluster.autoscaler.large.yaml screenshot</summary>
Screen Shot 2023-03-30 at 5 58 08 PM
</details>

<details>
<summary>ray-cluster.complete.large.yaml screenshot</summary>
Screen Shot 2023-03-30 at 5 33 24 PM
</details>

- [x] [ray-cluster.complete.large.yaml](https://github.com/ray-project/kuberay/blob/master/ray-operator/config/samples/ray-cluster.complete.large.yaml)
- [x] [ray-cluster.autoscaler.large.yaml](https://github.com/ray-project/kuberay/blob/master/ray-operator/config/samples/ray-cluster.autoscaler.large.yaml)

### User-facing Docs
- [x] [docker.md](https://github.com/ray-project/kuberay/blob/master/docs/deploy/docker.md)
- [x] [autoscaler.md](https://github.com/ray-project/kuberay/blob/master/docs/guidance/autoscaler.md) 

  ```sh
  kind create cluster --image=kindest/node:v1.23.0
  
  # My local branch I'm working on is currently at the latest commit of the cut branch `cd9b2e`.
  # Install the KubeRay operator from local Helm chart with the nightly Docker image (path: kuberay/helm-chart/kuberay-operator)
  helm install kuberay-operator . --set image.tag=nightly

  # Create an autoscaling RayCluster
  kubectl apply -f ray-cluster.autoscaler.yaml

  # Scale up the cluster
  export HEAD_POD=$(kubectl get pods -o custom-columns=POD:metadata.name | grep raycluster-autoscaler-head)
  kubectl exec $HEAD_POD -it -c ray-head -- python -c "import ray;ray.init();ray.autoscaler.sdk.request_resources(num_cpus=4)"
  ```
- [x] [development.md](https://github.com/ray-project/kuberay/blob/master/docs/development/development.md)
- [x] [manifests/README.md](https://github.com/ray-project/kuberay/blob/master/manifests/README.md) 
- [x] [helm-chart/ray-cluster/README.md](https://github.com/ray-project/kuberay/blob/master/helm-chart/ray-cluster/README.md): I haven't followed the instructions for installing a RayCluster. I only check for any updates required to move to version 0.5.0. Once version 0.5.0 is released, I will test this document again.
- [x] [helm-chart/kuberay-operator/README.md](https://github.com/ray-project/kuberay/blob/master/helm-chart/kuberay-operator/README.md): I haven't followed the instructions for installing a RayCluster. I only check for any updates required to move to version 0.5.0. Once version 0.5.0 is released, I will test this document again.
- [x] [kuberay-with-MCAD.md](https://github.com/ray-project/kuberay/blob/master/docs/guidance/kuberay-with-MCAD.md): We may need to improve the document. The document lacks information about how many resources should the Kubernetes cluster have to reproduce the experiment. In addition, it points to the [document](https://github.com/project-codeflare/multi-cluster-app-dispatcher/blob/quota-management/doc/usage/examples/kuberay/kuberay-mcad.md) in MCAD repo, and the document is really hard to follow if users do not have any context about MCAD. Open a PR https://github.com/project-codeflare/multi-cluster-app-dispatcher/pull/298 to fix some issues, and request the MCAD community to improve the document.
  ```sh
  kind create cluster --image=kindest/node:v1.23.0
  
  # Install MCAD
  # (path: https://github.com/project-codeflare/multi-cluster-app-dispatcher/tree/quota-management/deployment) 
  helm install mcad mcad-controller --namespace kube-system --set image.repository=quay.io/project-codeflare/mcad-controller,image.tag=release-v1.29.52
  
  # Update the ClusterRole created by the MCAD Helm chart to grant sufficient access for creating a RayCluster.
  # path: https://github.com/project-codeflare/multi-cluster-app-dispatcher/tree/quota-management/doc/usage/examples/kuberay/config
  kubectl apply -f xqueuejob-controller.yaml
  
  # Install KubeRay operator
  # path: kuberay/helm-chart/kuberay-operator
  helm install kuberay-operator . --set image.tag=nightly
  
  # Check the Kubernetes node's resource
  kubectl describe nodes kind-control-plane
  
  # CPU: (1) Request: 3050m (76%) (2) 2200m (55%)
  # I have 4 CPUs on my devbox, and I only remain 1 CPU at this moment. Hence, I updated `custompodresources` in `aw-raycluster.yaml` to 
  
  custompodresources:
  - replicas: 1
    requests:
      cpu: 600m
      ...
    limits:
      cpu: 600m
      ...
  
  # Create an AppWrapper
  # path: https://github.com/project-codeflare/multi-cluster-app-dispatcher/blob/quota-management/doc/usage/examples/kuberay/config/aw-raycluster.yaml
  kubectl apply -f aw-raycluster.yaml
  kubectl describe appwrappers.mcad.ibm.com raycluster-autoscaler | grep State
  # State:                         Running
  
  # However, if I update custompodresources's CPU request and limit to 2 and the names of the AppWrapper and RayCluster to `raycluster-autoscaler-2`, the RayCluster will not be dispatched.
  
  kubectl describe appwrappers.mcad.ibm.com raycluster-autoscaler-2 | grep State
  # State:                         Pending
  kubectl describe appwrappers.mcad.ibm.com raycluster-autoscaler-2 | grep Message
  # Message:                     Insufficient resources to dispatch AppWrapper.
  ```

- [x] [aws-eks-iam.md](https://github.com/ray-project/kuberay/blob/master/docs/guidance/aws-eks-iam.md): This issue is totally from Ray, and has no instruction from KubeRay, so I did not test this document. In addition, Ray 2.4.0 does not upgrade the version of boto3, so users may need to wait until Ray 2.5.0. See https://github.com/ray-project/ray/issues/33256 for more details.

- [x] [kubeflow-integration.md](https://github.com/ray-project/kuberay/blob/master/docs/guidance/kubeflow-integration.md)
   Screen Shot 2023-03-30 at 4 10 04 PM

- [x] [index.md](https://github.com/ray-project/kuberay/blob/master/docs/index.md)

- [x] [README.md](https://github.com/ray-project/kuberay/blob/master/README.md): I haven't followed the instructions for installing a RayCluster. I only check for any updates required to move to version 0.5.0. Once version 0.5.0 is released, I will test this document again.

- [x] [installation.md](https://github.com/ray-project/kuberay/blob/master/docs/deploy/installation.md): `single-namespace-resources` will install `cluster-scope-resources`, including a namespace `ray-system` and CRDs. However, only CRDs are needed.
   ```sh
   # Test: single-namespaced KubeRay
   kind create cluster --image=kindest/node:v1.23.0

   # Install CRD (path: kuberay/)
   kustomize build manifests/overlays/single-namespace-resources | envsubst | kubectl create -f -

   # Create a namespace "kuberay"
   export KUBERAY_NAMESPACE=kuberay
   kubectl create ns kuberay

   # Install the KubeRay operator in the namespace "kuberay" (path: kuberay/)
   kustomize build manifests/overlays/single-namespace | envsubst | kubectl apply -f -

   # Create a RayCluster in the "default" namespace (path: kuberay/ray-operator/config/samples)
   # No Pod will be generated.
   kubectl apply -f ray-cluster.mini.yaml

   # Create a RayCluster in the "kuberay" namespace (path: kuberay/ray-operator/config/samples)
   # Ray Pods will be created.
   kubectl apply -f ray-cluster.mini.yaml -n kuberay
   ```
- [x] [prometheus-grafana.md](https://github.com/ray-project/kuberay/blob/master/docs/guidance/prometheus-grafana.md): I skipped few steps in the doc (e.g. defining custom alert). 

  <details>
  <summary>Prometheus / Grafana screenshots</summary>
     Screen Shot 2023-03-30 at 5 33 24 PM
     Screen Shot 2023-03-30 at 5 33 24 PM
  </details>

### Helm charts

- [x] [kuberay-operator](https://github.com/ray-project/kuberay/tree/master/helm-chart/kuberay-operator)
   ```sh
   # path: helm-chart/kuberay-operator
   helm install kuberay-operator . --set image.tag=nightly
   ```

- [x] [ray-cluster](https://github.com/ray-project/kuberay/tree/master/helm-chart/ray-cluster)
  ```sh
  # path: helm-chart/kuberay-operator
  helm install kuberay-operator . --set image.tag=nightly
  # path: helm-chart/ray-cluster
  helm install ray-cluster .
  ```

### KubeRay operator (YAML)
```sh
kubectl create -k manifests/cluster-scope-resource
kubectl apply -k manifests/base
```
- [x] [ray-operator/config/crd](https://github.com/ray-project/kuberay/tree/master/ray-operator/config/crd)
- [x] [ray-operator/config/default](https://github.com/ray-project/kuberay/tree/master/ray-operator/config/default)
- [x] [ray-operator/config/manager](https://github.com/ray-project/kuberay/tree/master/ray-operator/config/manager) 
- [x] [ray-operator/config/rbac](https://github.com/ray-project/kuberay/tree/master/ray-operator/config/rbac) 
- [x] [manifests](https://github.com/ray-project/kuberay/tree/master/manifests) 

### Documents for KubeRay contributors
- [x] [DEVELOPMENT.md](https://github.com/ray-project/kuberay/blob/master/ray-operator/DEVELOPMENT.md): I do not follow the documents to build & test KubeRay. I only check if there are any updates required to move to version 0.5.0.

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by kevin85421** [comment]
Step 3.d in [release.md](https://github.com/ray-project/kuberay/blob/00dc45aec34fc3ae54962361221b41aaa4c338da/docs/development/release.md): Release Helm charts, and verify by the following commands.

```
helm repo add kuberay https://ray-project.github.io/kuberay-helm/
helm repo update
helm install kuberay-operator kuberay/kuberay-operator --version 0.5.0-rc.0
helm install kuberay-apiserver kuberay/kuberay-apiserver --version 0.5.0-rc.0
helm install raycluster kuberay/ray-cluster --version 0.5.0-rc.0
```

**Comment by kevin85421** [comment]
Step 3.d (for release v0.5.0) in [release.md](https://github.com/ray-project/kuberay/blob/00dc45aec34fc3ae54962361221b41aaa4c338da/docs/development/release.md): Release Helm charts, and verify by the following commands.

```sh
helm repo add kuberay https://ray-project.github.io/kuberay-helm/
helm repo update
helm search repo kuberay
# NAME                            CHART VERSION   APP VERSION     DESCRIPTION
# kuberay/kuberay-apiserver       0.5.0                           A Helm chart for kuberay-apiserver
# kuberay/kuberay-operator        0.5.0                           A Helm chart for Kubernetes
# kuberay/ray-cluster             0.5.0                           A Helm chart for Kubernetes

helm install kuberay-operator kuberay/kuberay-operator --version 0.5.0
helm install kuberay-apiserver kuberay/kuberay-apiserver --version 0.5.0
helm install raycluster kuberay/ray-cluster --version 0.5.0
```

**Review by architkulkarni — APPROVED** [review]
Looks good to me -- I went through everything but didn't do any extra testing
